accounts/abi/bind/backends/blockchain_test.go

```diff
@@ -147,3 +154,235 @@ func TestBlockchainCallContract(t *testing.T) {
+	}
+}
+
+func TestBlockChainChainID(t *testing.T) {
+	bc := newTestBlockchain()
+	c := NewBlockchainContractBackend(bc, nil, nil)
+
+	// Normal case
+	chainId, err := c.ChainID(context.Background())
+	assert.Nil(t, err)
+	assert.Equal(t, params.TestChainConfig.ChainID, chainId)
+}
+
+func initBackendForFiltererTests(t *testing.T, bc *blockchain.BlockChain) *BlockchainContractBackend {
+	block := bc.CurrentBlock()
+	state, _ := bc.State()
+
+	// Add one block with contract execution to generate logs
+	data_receive, _ := parsedAbi1.Pack("receive", []byte("X"))
+	tx := types.NewTransaction(uint64(0), code1Addr, big.NewInt(0), 50000, big.NewInt(1), data_receive)
+	chainId := bc.Config().ChainID
+
+	signedTx, err := types.SignTx(tx, types.LatestSignerForChainID(chainId), testKey)
+	if err != nil {
+		t.Errorf("could not sign tx: %v", err)
+	}
+
+	blocks, _ := blockchain.GenerateChain(bc.Config(), block, gxhash.NewFaker(), state.Database().TrieDB().DiskDB(), 1, func(i int, b *blockchain.BlockGen) {
+		b.AddTx(signedTx)
+	})
+	bc.InsertChain(blocks)
+
+	// mock filterer backend
+	mockCtrl := gomock.NewController(t)
+	mockBackend := mock_filter.NewMockBackend(mockCtrl)
+
+	any := gomock.Any()
+	txPoolConfig := blockchain.DefaultTxPoolConfig
+	txPoolConfig.Journal = "/dev/null" // disable journaling to file
+	txPool := blockchain.NewTxPool(txPoolConfig, bc.Config(), bc)
+	subscribeNewTxsEvent := func(ch chan<- blockchain.NewTxsEvent) klaytn.Subscription {
+		return txPool.SubscribeNewTxsEvent(ch)
+	}
+	subscribeLogsEvent := func(ch chan<- []*types.Log) klaytn.Subscription {
+		return bc.SubscribeLogsEvent(ch)
+	}
+	subscribeRemovedLogsEvent := func(ch chan<- blockchain.RemovedLogsEvent) klaytn.Subscription {
+		return bc.SubscribeRemovedLogsEvent(ch)
+	}
+	subscribeChainEvent := func(ch chan<- blockchain.ChainEvent) klaytn.Subscription {
+		return bc.SubscribeChainEvent(ch)
+	}
+	mockBackend.EXPECT().SubscribeNewTxsEvent(any).DoAndReturn(subscribeNewTxsEvent).AnyTimes()
+	mockBackend.EXPECT().SubscribeLogsEvent(any).DoAndReturn(subscribeLogsEvent).AnyTimes()
+	mockBackend.EXPECT().SubscribeRemovedLogsEvent(any).DoAndReturn(subscribeRemovedLogsEvent).AnyTimes()
+	mockBackend.EXPECT().SubscribeChainEvent(any).DoAndReturn(subscribeChainEvent).AnyTimes()
+
+	f := filters.NewEventSystem(&event.TypeMux{}, mockBackend, false)
+	c := NewBlockchainContractBackend(bc, f, nil)
+
+	return c
+}
+
+func TestBlockChainFilterLogs(t *testing.T) {
+	bc := newTestBlockchain()
+	c := initBackendForFiltererTests(t, bc)
+
+	// Normal case
+	logs, err := c.FilterLogs(context.Background(), klaytn.FilterQuery{
+		FromBlock: big.NewInt(10),
+		ToBlock:   big.NewInt(11),
+		Addresses: []common.Address{code1Addr},
+	})
+	assert.Nil(t, err)
+	assert.Equal(t, 2, len(logs))
+
+	// Non-code address
+	logs, err = c.FilterLogs(context.Background(), klaytn.FilterQuery{
+		FromBlock: big.NewInt(0),
+		ToBlock:   big.NewInt(11),
+		Addresses: []common.Address{testAddr},
```

Comment:
How about to change with code2Addr and the comment (L334) as // No receipt logs exist for code1Addr`

Reply:
Updated in c2b0f3d, thanks!